src/sagemaker/predictor.py

```diff
@@ -105,9 +105,25 @@ def _create_request_args(self, data, initial_args=None):
         args['Body'] = data
         return args
 
-    def delete_endpoint(self):
-        """Delete the Amazon SageMaker endpoint backing this predictor.
+    def _delete_endpoint_config(self):
+        """Delete the Amazon SageMaker endpoint configuration
         """
+        endpoint_description = self.sagemaker_session.sagemaker_client.describe_endpoint(EndpointName=self.endpoint)
+        endpoint_config_name = endpoint_description['EndpointConfigName']
+        self.sagemaker_session.delete_endpoint_config(endpoint_config_name)
+
+    def delete_endpoint(self, delete_endpoint_config=True):
+        """Delete the Amazon SageMaker endpoint backing this predictor. Also delete the endpoint configuration attached
```

Comment:
I think it is okay to say:

"Delete the Amazon SageMaker endpoint and endpoint configuration backing this predictor."

Since it is deleted by default.